signer/rpc_and_client_test.go

```diff
@@ -80,73 +81,152 @@ func (s stubServer) Sign(ctx context.Context, sr *pb.SignatureRequest) (*pb.Sign
-	s := getStubbedHealthServer(func() (map[string]string, error) {
-		return map[string]string{"health": "not good"}, nil
-	})
+// healthCheckUnhealthy does not succeed if the service is unhealthy
+func healthCheckUnhealthy(t *testing.T, serviceName string) {
+	hs := health.NewServer()
+	hs.SetServingStatus(serviceName, healthpb.HealthCheckResponse_NOT_SERVING)
+
+	s := getStubbedHealthServer(hs)
 	signerClient, _, cleanup := setUpSignerClient(t, s)
 	defer cleanup()
-	require.Error(t, signerClient.CheckHealth(1*time.Second))
+	require.Error(t, signerClient.CheckHealth(1*time.Second, serviceName))
 }
 
-// CheckHealth does not succeed if the health check to the KM server errors
-func TestHealthCheckKMError(t *testing.T) {
-	s := getStubbedHealthServer(func() (map[string]string, error) {
-		return nil, errors.New("Something's wrong")
-	})
-	signerClient, _, cleanup := setUpSignerClient(t, s)
-	defer cleanup()
-	require.Error(t, signerClient.CheckHealth(1*time.Second))
+// TestHealthCheckKMUnhealthy does not succeed if the KM server is unhealthy
+func TestHealthCheckKMUnhealthy(t *testing.T) {
+	healthCheckUnhealthy(t, notary.HealthCheckKeyManagement)
 }
 
-// CheckHealth does not succeed if the health check to the KM server times out
-func TestHealthCheckKMTimeout(t *testing.T) {
-	s := getStubbedHealthServer(func() (map[string]string, error) {
-		return nil, fmt.Errorf("this should should have failed by now")
-	})
+// TestHealthCheckSignerUnhealthy does not succeed if the Signer server is unhealthy
+func TestHealthCheckSignerUnhealthy(t *testing.T) {
+	healthCheckUnhealthy(t, notary.HealthCheckSigner)
+}
+
+// healthCheckTimeout does not succeed if the health check to the server times out
+func healthCheckTimeout(t *testing.T, serviceName string) {
+	hs := health.NewServer()
+	hs.SetServingStatus(serviceName, healthpb.HealthCheckResponse_NOT_SERVING)
+
+	s := getStubbedHealthServer(hs)
 	signerClient, _, cleanup := setUpSignerClient(t, s)
 	defer cleanup()
 
-	err := signerClient.CheckHealth(0 * time.Second)
+	err := signerClient.CheckHealth(0*time.Second, serviceName)
 	require.Error(t, err)
-	require.True(t, strings.Contains(err.Error(), "Timed out"))
+	require.Contains(t, err.Error(), context.DeadlineExceeded.Error())
 }
 
-// CheckHealth succeeds if KM is healthy and reachable.
-func TestHealthCheckKMHealthy(t *testing.T) {
-	s := getStubbedHealthServer(func() (map[string]string, error) {
-		return nil, nil
-	})
+// TestHealthCheckKMTimeout does not succeed if the health check to the KM server times out
+func TestHealthCheckKMTimeout(t *testing.T) {
+	healthCheckTimeout(t, notary.HealthCheckKeyManagement)
+}
+
+// TestHealthCheckSignerTimeout does not succeed if the health check to the Signer server times out
+func TestHealthCheckSignerTimeout(t *testing.T) {
+	healthCheckTimeout(t, notary.HealthCheckSigner)
+}
+
+// healthCheckHealthy succeeds if server is healthy and reachable.
+func healthCheckKMHealthy(t *testing.T, serviceName string) {
```

Comment:
nit: function name should probably be `healthCheckHealthy` to match the description